### PR TITLE
Transfer monitor might not known destination of the file being uploaded.

### PR DESCRIPTION
**Pull request by HelclM**
This happens in case of POST from PrusaLink.

### CODE REVIEW

File: src/connect/render.cpp [hunk 1]
```diff
@@ -348,7 +348,9 @@ namespace {
                     JSON_FIELD_INT_G(transfer_status.has_value(), "size", transfer_status->expected) JSON_COMMA;
                     JSON_FIELD_INT_G(transfer_status.has_value(), "transferred", transfer_status->transferred) JSON_COMMA;
                     JSON_FIELD_INT_G(transfer_status.has_value(), "time_transferring", (transfer_status->start - ticks_ms()) / 1000) JSON_COMMA;
-                    JSON_FIELD_STR_G(transfer_status.has_value(), "path", transfer_status->destination) JSON_COMMA;
+                    if (transfer_status->destination) {
+                        JSON_FIELD_STR_G(transfer_status.has_value(), "path", transfer_status->destination) JSON_COMMA;
+                    }
```
Comment (vorner) [inline src/connect/render.cpp:353]: I agree this'll actually work in practice, but is a bit fragile and non-trivial to prove (and is based on the fact destination never changes from non-null to null).

I think the reason _why_ this works needs a comment.

File: src/transfers/monitor.hpp [hunk 2]
```diff
@@ -208,6 +212,8 @@ class Monitor {
     Timestamp start;
     size_t expected;
     size_t transferred;
+
+    bool path_known = false;
```
Comment (vorner) [inline src/transfers/monitor.hpp:216]: Is it worth the effort, or are we OK conflating empty path and null path here?

I guess `optional<char [FILE_PATH_BUFFER_LEN]>` would be too much of a mental exercise? 😈

**Review by vorner — COMMENTED** [review]
Minor nitpicks, probably worth addressing. But it is something you can safely build other things on without fear it'll change, in case you happen to return to work before me 😇

File: src/transfers/monitor.hpp (hunk 2 above)
Comment (HelclM) [reply src/transfers/monitor.hpp:216]: I agree we could just omit the bool, but AFAIK you can't put an array type into a optional. (taken from [cppreference](https://en.cppreference.com/w/cpp/utility/optional))
image

**Review by vorner — APPROVED** [review]

**Pull request merged**